lib/ash_authentication.ex

```diff
@@ -171,42 +103,40 @@ defmodule AshAuthentication do
   @doc """
   Find all resources which support authentication for a given OTP application.
 
-  Returns a map where the key is the authentication provider, and the values are
-  lists of api/resource pairs.
+  Returns a list of resource modules.
+
+  ## Example
+
+      iex> authenticated_resources(:ash_authentication)
+      [Example.User]
 
-  This is primarily useful for introspection, but also allows us to simplify
-  token lookup.
   """
-  @spec authenticated_resources(atom) :: [resource_config]
+  @spec authenticated_resources(atom) :: [Resource.t()]
   def authenticated_resources(otp_app) do
     otp_app
     |> Application.get_env(:ash_apis, [])
     |> Stream.flat_map(&Api.Info.resources(&1))
-    |> Stream.map(&resource_config/1)
-    |> Stream.reject(&(&1 == :error))
+    |> Stream.filter(&(AshAuthentication in Spark.extensions(&1)))
     |> Enum.to_list()
   end
 
-  def resource_config(resource) do
-    resource
-    |> Extension.get_persisted(:authentication)
-    |> case do
-      nil ->
-        :error
+  @doc """
+  Return a subject string for user.
 
-      config ->
-        Map.put(config, :resource, resource)
-    end
-  end
+  This is done by concatenating the resource's subject name with the resource's
+  primary key field(s) to generate a uri-like string.
+
+  Example:
+
+      iex> build_user(id: "ce7969f9-afa5-474c-bc52-ac23a103cef6") |> user_to_subject()
+      "user?id=ce7969f9-afa5-474c-bc52-ac23a103cef6"
 
-  @doc """
-  Return a subject string for an AshAuthentication resource.
   """
-  @spec resource_to_subject(Resource.record()) :: subject
-  def resource_to_subject(record) do
+  @spec user_to_subject(Resource.record()) :: subject
+  def user_to_subject(record) do
```

Comment:
Should we be using the term `user` here? I think we may want to use the term `actor`, since theoretically something like a service account could could be authenticating.

Reply:
I went back and forwards on this a few times, but ultimately I went back to `user` as being the common case and less confusing than other options - most of which already have meanings in Ashland.

Reply:
Talked off line, agreed to stick with user as a specific concept to `ash_authentication`. i.e "its not an actor until you use it as an actor in your ash resources".